src/Raven.Server/Documents/ETL/Providers/Queue/AwsSqs/AwsSqsEtl.cs

```diff
@@ -0,0 +1,270 @@
+        BlittableJsonEventBinaryFormatter formatter, out List<string> idsToDelete)
+    {
+        if (itemsPerQueue.Count == 0)
+        {
+            idsToDelete = null;
+            return 0;
+        }
+
+        var tooLargeDocsErrors = new Queue<EtlErrorInfo>();
+        idsToDelete = new List<string>();
+        int count = 0;
+
+        foreach (QueueWithItems<AwsSqsItem> queue in itemsPerQueue)
+        {
+            string queueName = queue.Name.ToLower();
+
+            if (_queueClient == null)
+            {
+                _queueClient = QueueBrokerConnectionHelper.CreateAwsSqsClient(
+                    Configuration.Connection.AwsSqsConnectionSettings);
+            }
+
+            if (Configuration.SkipAutomaticQueueDeclaration == false &&
+                _alreadyCreatedQueues.ContainsKey(queueName) == false)
+                AsyncHelpers.RunSync(() => CreateQueue(_queueClient, queueName));
+
+            var batchMessages = new List<SendMessageBatchRequestEntry>();
+
+            foreach (AwsSqsItem queueItem in queue.Items)
+            {
+                CancellationToken.ThrowIfCancellationRequested();
+
+                try
+                {
+                    var sendMessageEntry = new SendMessageBatchRequestEntry
+                    {
+                        Id = queueItem.DocumentId,
+                        MessageBody = SerializeCloudEvent(queueItem)
+                    };
+
+                    batchMessages.Add(sendMessageEntry);
+
+                    if (batchMessages.Count == 10)
+                    {
+                        SendBatchMessages(queueName, batchMessages, queue, idsToDelete, tooLargeDocsErrors);
+                        count += batchMessages.Count;
+                        batchMessages.Clear();
+                    }
+                }
+                catch (Exception ex)
+                {
+                    throw new QueueLoadException($"Failed to prepare message, error reason: '{ex.Message}'", ex);
+                }
+            }
+
+            if (batchMessages.Count > 0)
+            {
+                SendBatchMessages(queueName, batchMessages, queue, idsToDelete, tooLargeDocsErrors);
+                count += batchMessages.Count;
+                batchMessages.Clear();
+            }
+
+            if (tooLargeDocsErrors.Count > 0)
+            {
+                Database.NotificationCenter.EtlNotifications.AddLoadErrors(Tag, Name, tooLargeDocsErrors,
+                    "ETL has partially loaded the data. " +
+                    "Some of the documents were too big (>256KB) to be handled by Aws Sqs. " +
+                    "It caused load errors, that have been skipped. ");
+            }
+        }
+
+        return count;
+    }
+
+    private void SendBatchMessages(string queueName, List<SendMessageBatchRequestEntry> batchMessages,
+        QueueWithItems<AwsSqsItem> queue, List<string> idsToDelete, Queue<EtlErrorInfo> tooLargeDocsErrors)
```

Comment:
Can we mark `idsToDelete` and `tooLargeDocsErrors` as `ref` so it will be more obvious that they are not just input parameters but they are modified inside.

Reply:
fixed